### PR TITLE
fix (service): reload telegraf if config file was changed

### LINKED ISSUE #5: Reload telegraf if config file was changed
Currently telegraf isn't reloaded if the config file was changed. That seems to me not really practically.

### PR DESCRIPTION
**Pull request by herbetom**
fixes #5

### CODE REVIEW

File: telegraf/service/running.sls [hunk 1]
```diff
@@ -15,3 +15,6 @@ telegraf/service/running:
     - enable: True
     - require:
       - sls: {{ sls_config_file }}
+    - reload: True
+    - watch:
+      - file: {{ sls_config_file }}
```
Comment (myii) [inline telegraf/service/running.sls:20]:
```suggestion
      - sls: {{ sls_config_file }}
```

`sls_config_file` is the whole `sls`.

Comment (herbetom) [reply telegraf/service/running.sls]: I really don't now. I added since it worked for me.

**Comment by herbetom** [comment]
@myii I have to say i was a bit frustated earlier. Sorry for that. That's also the reason why i didn't fill the template.

I made the changes

File: telegraf/service/running.sls (hunk 1 above)
Comment (herbetom) [reply telegraf/service/running.sls]: > When set, reload the service instead of restarting it.

**Comment by myii** [comment]
> @myii I have to say i was a bit frustated earlier. Sorry for that. That's also the reason why i didn't fill the template.

Understood, that goes for all of us.  It's important to note that we're all doing this in our free time, so that's why it's important to help each other out with the details.

> I made the changes

Appreciate it, I'm just waiting for the tests to complete and then I'll get this merged.

By the way, I've just tested this formula against Salt's `master` branch (leading up to Sodium `3001`) and it is failing because of changes to the `toml` module (https://travis-ci.org/github/myii/telegraf-formula/builds/681360474).  I've informed the community in the #testing channel in Slack so there'll be some discussions about that before we make the changes here.  Just letting you know so that you can get frustrated in advance!

**Comment by myii** [comment]
@herbetom Just realised that we've got some of the `master` branch images being tested here, that were built this morning (so have the Sodium changes in them).  This PR is fine, so I've merged it.  I'll change the Travis matrix soon.  Thanks again for your contribution, responses to the reviews and your patience!

**Comment by herbetom** [comment]
Thanks a lot! I appreciate the work here. My frustration had to do with something else.

**Comment by myii** [comment]
> By the way, I've just tested this formula against Salt's `master` branch (leading up to Sodium `3001`) and it is failing because of changes to the `toml` module (https://travis-ci.org/github/myii/telegraf-formula/builds/681360474). I've informed the community in the #testing channel in Slack so there'll be some discussions about that before we make the changes here. Just letting you know so that you can get frustrated in advance!

@herbetom If #7 gets merged, then you won't need to worry about this anymore!

**Comment by herbetom** [comment]
Thanks a lot! Hopefully that fixes it 👍

Now that I think about it, when using Salt 3000.2 I ran into some toml error when using it with `salt 'test1*' state.apply test=True` (when the config file wasn't present). I'm curious if that fixes that.

**Comment by myii** [comment]
> Now that I think about it, when using Salt 3000.2 I ran into some toml error when using it with `salt 'test1*' state.apply test=True` (when the config file wasn't present). I'm curious if that fixes that.

@herbetom `3000.2` is still using `pytoml`, so you'll need the relevant `python*-pytoml` package for your OS, as this formula already provides once you run the `package/install` states.

https://github.com/saltstack-formulas/telegraf-formula/blob/5d33d2b16b6ff0338fbb211972bccdc17922ce46/telegraf/map.jinja#L51-L65

The change made in upstream Salt has been [merged in for Sodium](https://github.com/saltstack/salt/pull/56615/files#diff-86cde5c3ade9feb1ab1bd1b7b67772bfL21-R21) (i.e. the current `master` branch).

**Comment by herbetom** [comment]
Ah, that's explains it and makes sense. Before you wrote that, i was trying to recreate that error, but since i only purged telegraf it was working correctly. 

Thanks for explaining that to me!

**Pull request merged**